### ISSUE TITLE
[New Feature] Vehicles and even infantry consume electricity, and the corresponding consequences are allowed

### ISSUE THREAD

**Issue by STstarstoneZ** [Phobos-developers/Phobos#455]
I hope unit can consume the same power as buildings, when low power, unit can also get the corresponding effect. For example, unable to move, self damage, invalid self effect, or reduced attack and other AE effects.

[TechnoType]
Poweruse=(integer - power)

LP.Operate=(boolean)

LP.SelfHealing=(boolean)
LP.SelfHealing.Rate= (double - minutes)
LP.SelfHealing.Amount= (integer - hitpoints)

LP.Cloakable= (boolean) 
LP.SensorArray= (boolean) 

LP.Effect=(AnimationType)

**Comment by 4SGX** [comment]
Might as well proliferate `Power` instead of creating Poweruse so it can be positive or negative.

**Comment by STstarstoneZ** [comment]
yes, you are right. When the vehicle use power, it is also reasonable to generate power.
